wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -629,7 +629,7 @@ public TypeSpec generateMessage(MessageType type) {
    * Decides if a constructor should take all fields or a builder as a parameter.
    */
   private boolean constructorTakesAllFields(MessageType type) {
-    return type.fields().size() < MAX_PARAMS_IN_CONSTRUCTOR;
+    return type.fieldsAndOneOfFields().size() < MAX_PARAMS_IN_CONSTRUCTOR;
```

Comment:
Does this fix an old bug? Looks out of scope of this PR, and needs a test

Reply:
yes it did, will move it out.